### DISCUSSION TITLE
How to use argo-rollouts apiClient

### DISCUSSION THREAD

**Discussion by JuozasVainauskas** [argoproj/argo-rollouts#2799] [Q&A]
Hello,

I want to use [argo-rollout apiClient](https://github.com/argoproj/argo-rollouts/tree/master/pkg/apiclient) to interact with argo-rollouts resources from my Golang application (in this case, `promote` rollout in certain scenarios). However, I am not entirely sure how to use argo-rollouts [apiClient](https://github.com/argoproj/argo-rollouts/tree/master/pkg/apiclient).

For example, in [ArgoCD](https://github.com/argoproj/argo-cd/blob/master/pkg/apiclient/apiclient.go) and [argo-workflows](https://github.com/argoproj/argo-workflows/blob/master/pkg/apiclient/apiclient.go) cases I am able to use `pkg/apiclient/apiclient.go` to create clients for particular resources (e.g. [ApplicationClient](https://github.com/argoproj/argo-cd/blob/87a7dff4585c514304c9e46cdbe8ad20051c935f/pkg/apiclient/apiclient.go#L91), [WorkflowServiceClient](https://github.com/argoproj/argo-workflows/blob/2a5c4e8d2c983b88044c84e90e0aa5805edfe5b2/pkg/apiclient/apiclient.go#L21), etc.)

In argo-rollouts case, I want to create [RolloutServiceClient](https://github.com/argoproj/argo-rollouts/blob/6ac153326c585a79555e3c5f5cfec0b22fce1f61/pkg/apiclient/rollout/rollout.pb.go#L1832). However, `pkg/apiclient/apiclient.go` file for creation of the client is not present in the package. I assume that this file should be available in the package and that this is a bug (If this is the case, I will create an issue). However, if I am mistaken and it is indeed intentional, how should I create a `RolloutServiceClient`?

**Comment by YannickZ** [comment]
Did you find any answers to this @JuozasVainauskas ?

**Reply by JuozasVainauskas** [reply]
Hi, I think the question was answered in this thread: https://github.com/argoproj/argo-rollouts/issues/2815

**Reply by YannickZ** [reply]
i saw that, but did you find any way to do this ?

**Reply by JuozasVainauskas** [reply]
I think the only solution for this is directly adjusting k8s resources.
